### DISCUSSION TITLE
What is this cool hover effect used on Nuxt.com website for the grid?

### DISCUSSION THREAD

**Discussion by martinszeltins** [nuxt/nuxt#23772] [Questions]
What is this CSS effect? Is it using some library to achieve this?

![image](https://github.com/nuxt/nuxt/assets/34019878/60c9a4e2-5ae7-42c5-8210-20798893d3d6)

**Comment by Jannchie** [comment]
It seems that https://ui.nuxt.com/pro implements this effect... Doesn't look like it's completely open source.

This doesn't seem too difficult to implement, the border is actually a background gradient stacked on top of the back, and the appearance of the gradient is changed by monitoring the mouse position (which can be done using vueuse) and dynamically setting css variables.

**Reply by madc** [reply]
Here's a direct link: https://ui.nuxt.com/pro/components/landing/landing-grid

**Comment by Aleroms** [comment]
I sort of implemented a version of it cause I thought it was cool as well. Basically, find the center of element, create x and y vectors, that way you can degree (depending on mouse pos) to use in linear gradient as mentioned above.
```js
//src/composables/nuxtHoverEffect.js
import { computed } from 'vue'
import { useMouseInElement } from '@vueuse/core'
import { useThemeStore } from '@/stores/theme.js'

function lerp(start, end, factor) {
  return start * (1 - factor) + end * factor
}

export function useNuxtHoverEffect(target) {
  const { elementX, elementY, isOutside, elementHeight, elementWidth } = useMouseInElement(target)
  const theme = useThemeStore()
  const nuxtHoverEffect = computed(() => {
    //calculate center
    let centerX = elementWidth.value / 2
    let centerY = elementHeight.value / 2
    //calc vectors from center to mouse pos
    let vectorX = centerX - elementX.value
    let vectorY = elementY.value - centerY
    let vectorLength = Math.sqrt(vectorX ** 2 + vectorY ** 2)
    //calc angle
    let angleRad = Math.atan2(vectorX, vectorY)
    //convert to deg
    let deg = angleRad * (180 / Math.PI)
    // calculate opacity based on vectorLength
    let opacity = lerp(1, 0, vectorLength / 600)

    //colors
    let solid_color = !theme.isDark ? 'rgba(57,181,74,1)' : 'rgba(30,83,219,1)'
    let highlight = !theme.isDark
      ? `linear-gradient(${deg}deg, rgba(57, 181, 74,${opacity}), rgba(57, 181, 74,0) 20%)`
      : `linear-gradient(${deg}deg, rgba(30, 83, 219,${opacity}), rgba(30,83,219,0) 20%)`
    return isOutside.value ? highlight : solid_color
  })

  return {
    target,
    elementX,
    elementY,
    isOutside,
    elementHeight,
    elementWidth,
    nuxtHoverEffect
  }
}
```
I am using the composable in my [NuxtHoverEffect.vue](https://github.com/Aleroms/santiagomorales/blob/main/src/components/Utilities/NuxtHoverEffect.vue)
It is not a 1 to 1 match though.

**Reply by martinszeltins** [reply]
Cool, can you create a live demo of this?

**Reply by Aleroms** [reply]
already did on my portfolio, here's the [link](https://santiagomorales.dev/projects). It's not perfect but I'm sure it will be a nice starter in case someone can improve it.

**Reply by martinszeltins** [reply]
@Aleroms that's a good start, it's not perfect but it's in the right direction. Thanks for sharing!

**Comment by arimaulana07** [comment]
Im implemented this effect on my web portofolio.

check live demo on StackBlitz here: https://stackblitz.com/edit/nuxt-starter-x4xz6w?file=components%2FTheBox.vue
git repo: https://github.com/arimaulana07/card-hover-effect

with the help of vueuse, implementing this effect is so easy. You just need place the radial gradient based on your mouse position (which you can get with useMouseInElement()).

**Comment by BenHerbst** [comment]
I created a tutorial on how to create this effect 1 by 1:
https://www.devsfordevs.com/blogs/96-Nuxt-Like-CSS-Effect

**Reply by Gmiller123** [reply]
by any chance you can write the same code for react

**Comment by dimar-hanung** [comment]
I improving little from @BenHerbst code, now have multiple card and rounded.
Live Demo: https://dimarhanung.dev/fun-stuff/card-hover
Code: https://github.com/dimar-hanung/dimarhanung-next/blob/main/pages/fun-stuff/card-hover.vue .
PR are very welcome 😁
![image](https://github.com/nuxt/nuxt/assets/42312636/c6fb4c65-bb6b-4e6b-9234-ae55a99bc701)

**Comment by antiDave** [comment]
I updated this, it works with standard html, css, and js. I fixed the bottom as well. When i have more time i will update to add the original border also.
https://github.com/antiDave/nuxt_reproduction
